src/werkzeug/utils.py

```diff
@@ -114,7 +114,8 @@ def invalidate_cached_property(obj, name):
 
     You must pass the name of the cached property as the second argument.
     """
-    obj.__dict__[name] = _missing
+    if isinstance(getattr(obj.__class__, name, None), cached_property):
```

Comment:
maybe `if not isinstance(...): raise TypeError(...)`

Reply:
Is raising an exception necessary? Can't we simply issue a warning to stderr?

Reply:
I think it should raise an error, if it's called on a non-cached_property it's always going to do the wrong thing and is always a programming error

Reply:
Ok, updated.